### PR TITLE
 refactor: refactor to use navigation ACTIONS API

### LINKED ISSUE #1614: feat: Use the Actions API from Nav AAC for navigating between screens
**Describe the feature you'd like**
The Navigation architecture component has an Actions API to identify navigation paths within an app. Navigating using actions allows for a lot cleaner code, and also improves backstack management because the Nav AAC can then figure out nested layers of our app.

**Additional context**
This is going to be fairly large change, so I plan on doing this in multiple commits. I shall squash all the commits into one before finalizing the PR.

**Would you like to work on the issue?**
- [x] Yes

### PR DESCRIPTION
**Pull request by aggarwalpulkit596**
Fixes #1614 

Changes: Refractor app to navigation global actions

Screenshots for the change:

### CODE REVIEW

**Comment by aggarwalpulkit596** [comment]
@haroldadmin I hope you wouldn't mind that I completed this task

**Comment by aggarwalpulkit596** [comment]
@nikit19 @liveHarshit please review

File: app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt [hunk 1]
```diff
@@ -124,23 +124,6 @@ object Utils {
...
-    }
-
-    fun getAnimSlide(): NavOptions {
-        val builder = NavOptions.Builder()
-        builder.setEnterAnim(R.anim.slide_in_right)
-        builder.setExitAnim(R.anim.slide_out_left)
-        builder.setPopEnterAnim(R.anim.slide_in_left)
-        builder.setPopExitAnim(R.anim.slide_out_right)
-        return builder.build()
-    }
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: I don't think these should remove. It can be used later for any fragment navigation.

File: app/src/main/java/org/fossasia/openevent/general/search/SearchTypeFragment.kt [hunk 2]
```diff
@@ -8,12 +8,11 @@ import android.view.ViewGroup
 import androidx.core.view.isVisible
 import androidx.fragment.app.Fragment
 import androidx.lifecycle.Observer
-import androidx.navigation.NavOptions
 import androidx.navigation.Navigation
 import androidx.navigation.fragment.navArgs
 import androidx.recyclerview.widget.LinearLayoutManager
-import kotlinx.android.synthetic.main.content_no_internet.view.retry
 import kotlinx.android.synthetic.main.content_no_internet.view.noInternetCard
+import kotlinx.android.synthetic.main.content_no_internet.view.retry
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/search/SearchTypeFragment.kt]: Unrelated changes.

**Comment by liveHarshit** [comment]
Build is also failing.

**Comment by aggarwalpulkit596** [comment]
@liveHarshit updated and can you also check if there is any wrong navigation flow because it's a large pr and I might miss one or two

**Comment by liveHarshit** [comment]
> @liveHarshit updated and can you also check if there is any wrong navigation flow because it's a large pr and I might miss one or two

You should check it by installing the app in a device and testing each fragment by opening them and going back. Inform when all is okay.

File: app/src/main/java/org/fossasia/openevent/general/search/SearchTypeFragment.kt (hunk 2 above)
Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchTypeFragment.kt]: And that"?

File: app/src/main/java/org/fossasia/openevent/general/search/SearchTypeFragment.kt [hunk 3]
```diff
@@ -28,7 +28,6 @@ class SearchTypeFragment : Fragment() {
     private val safeArgs: SearchTypeFragmentArgs by navArgs()
     private lateinit var rootView: View
     private val eventTypesList: MutableList<String> = arrayListOf("Anything")
-
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/search/SearchTypeFragment.kt]: Do not remove newline before function.

File: app/src/main/java/org/fossasia/openevent/general/search/SearchTypeFragment.kt [hunk 4]
```diff
@@ -74,7 +73,6 @@ class SearchTypeFragment : Fragment() {
                 typesAdapter.addAll(eventTypesList)
                 typesAdapter.notifyDataSetChanged()
             })
-
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/search/SearchTypeFragment.kt]: same

File: app/src/main/res/navigation/navigation_graph.xml [hunk 5]
```diff
@@ -9,18 +9,57 @@
...
+
+    </fragment>
     <fragment
         android:id="@+id/similarEventsFragment"
         android:name="org.fossasia.openevent.general.event.topic.SimilarEventsFragment"
         android:label="SimilarEventsFragment"
         tools:layout="@layout/fragment_similar_events">
 
+        <action
+            android:id="@+id/action_similarEvents_to_eventDetails"
```
Comment (nikit19) [inline app/src/main/res/navigation/navigation_graph.xml]: fusion of snake case and camel case?

Comment (liveHarshit) [reply app/src/main/res/navigation/navigation_graph.xml]: @aggarwalpulkit596 Please use snake case only according to code style guideline.

Comment (aggarwalpulkit596) [reply app/src/main/res/navigation/navigation_graph.xml]: @nikit19 @liveHarshit will update them

**Comment by aggarwalpulkit596** [comment]
@nikit19 @liveHarshit updated

**Comment by haroldadmin** [comment]
You are not using Actions to their full extent. When navigating from one destination to the other, you are passing the ID of the action. Instead, you should use the generated FragmentDirections class to get an instance of the action object, pass it the arguments using the FragmentArgs generated class, and then navigate to it.

Something like:
```
EventsFragmentDirections
  .viewEventDetails(eventId = event.id)
  .let { action ->
    findNavController().navigate(action)
  }
```

**Comment by haroldadmin** [comment]
Also, please use sensible names for the actions. The name generated by Android Studio is a generic name, and does not convey any information regarding what the action is supposed to do. 

I'd also suggest using camel case for event names instead of snake case, since they are interacted with in Kotlin code. Although this is just a style preference.

**Comment by aggarwalpulkit596** [comment]
> Also, please use sensible names for the actions. The name generated by Android Studio is a generic name, and does not convey any information regarding what the action is supposed to do.
> 
> I'd also suggest using camel case for event names instead of snake case, since they are interacted with in Kotlin code. Although this is just a style preference.

I haven't generated action names using android studio I manually wrote them so I think they're sensible enough like
actionAttendeeToOrderCompleted is quite sensible i guess

**Comment by aggarwalpulkit596** [comment]
@iamareebjamal @haroldadmin review

**Comment by liveHarshit** [comment]
@iamareebjamal Please review it.

File: app/src/fdroid/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt [hunk 6]
```diff
@@ -23,7 +22,9 @@ class WelcomeFragment : Fragment() {
         rootView.currentLocation.visibility = View.GONE
 
         rootView.pickCityButton.setOnClickListener {
-            Navigation.findNavController(rootView).navigate(R.id.searchLocationFragment, null, Utils.getAnimSlide())
+            WelcomeFragmentDirections.actionWelcomeToSearch().let {
+                findNavController().navigate(it)
+            }
```
Comment (iamareebjamal) [inline app/src/fdroid/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt]: Why not `findNavController().navigate(WelcomeFragmentDirections.actionWelcomeToSearch())?`

Comment (aggarwalpulkit596) [reply app/src/fdroid/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt]: @iamareebjamal For ease of adding variables if added future

Comment (iamareebjamal) [reply app/src/fdroid/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt]: IDK how will it make it easier. Give an example

Comment (aggarwalpulkit596) [reply app/src/fdroid/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt]:
```
 WelcomeFragmentDirections.actionWelcomeToSearch(query).let {
                findNavController().navigate(it)
            }
```
findNavController().navigate(WelcomeFragmentDirections.actionWelcomeToSearch(
query = "",
            location = Preference().getString(SAVED_LOCATION).toString(),	            location = Preference().getString(SAVED_LOCATION).toString(),
            type = getString(R.string.anything),	            type = getString(R.string.anything),
            date = time)	            date = time


))

Comment (aggarwalpulkit596) [reply app/src/fdroid/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt]: @iamareebjamal the first one look much nice in terms of viewability although it is just a preference hardly made a difference

Comment (iamareebjamal) [reply app/src/fdroid/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt]: I think it looks more verbose. There is no use of let here. When you add more arguments to first method, it'll look just as bad

Comment (aggarwalpulkit596) [reply app/src/fdroid/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt]: I'll refractor it

**Comment by aggarwalpulkit596** [comment]
@iamareebjamal updated
